source/features/restore-file.tsx

```diff
@@ -79,32 +74,47 @@ async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButt
 
 	filesRestored.add(menuItem);
 
-	menuItem.textContent = 'Restoring…';
-	event.preventDefault();
-	event.stopPropagation();
-
-	try {
-		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
+	async function restoreFile(filePath: string, progress: (message: string) => void): Promise<void> {
+		progress('Restoring...');
 		const file = await getFile(filePath);
 
 		if (!file) {
 			// The file was created by this PR.
 			// This code won’t be reached if `highlight-deleted-and-added-files-in-diffs` works.
-			showError(menuItem, 'Nothing to restore. Delete file instead');
-			return;
+			throw new Error('Nothing to restore. Delete file instead');
 		}
 
 		if (file.isTruncated) {
-			showError(menuItem, 'Restore failed: File too big');
-			return;
+			throw new Error('Restore failed: File too big');
 		}
 
-		await commitFileContent(menuItem, file.text, filePath);
+		await commitFileContent(menuItem, file.text, filePath, progress);
+	}
 
+	try {
+		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
+		// Show toast while restoring
+		const progress = async (message: string): Promise<void> => {
```

Comment:
Not what I meant, refer to my previous example exactly. `progress` is a callback provided by showToast itself to allow its callback to change the text. 

See the line `return await task()`? It should be

```JavaScript 
return await task(message => {
   messageWrapper.textContent = message
})
```

Reply:
Could you please elaborate further on this?

Reply:
I don't know what else to say. I gave the exact code to paste there and how to use it in my previous comment.

Reply:
Trying to make this work, Unsuccessful so far.